src/kafka-util/src/client.rs

```diff
@@ -280,8 +308,80 @@ where
+                        // join it so that the rdkafka thread calling this callback
+                        // spends all its time parked, instead of being
+                        // continuously woken up when we `block_on` the
+                        // ssh tunnel creation.
+                        //
+                        // Note that we hold the async lock while we create this tunnel; this is to
+                        // prevent competing threads from thrashing the ssh bastion host.
+                        let ssh_tunnel = match std::thread::spawn(move || {
+                            runtime.block_on(async move {
+                                ssh_tunnel_manager
+                                    .connect(
+                                        tunnel_id,
+                                        default_tunnel,
+                                        &broker.host,
+                                        broker.port.parse().unwrap(),
+                                    )
+                                    .await
+                            })
+                        })
+                        .join()
+                        {
+                            Ok(ssh_tunnel) => ssh_tunnel,
+                            Err(e) => std::panic::resume_unwind(e),
+                        };
+
+                        match ssh_tunnel {
+                            Ok(ssh_tunnel) => {
+                                let rewrite_tunnel = ssh_tunnel.clone();
+                                rewrites.insert(
+                                    addr.clone(),
+                                    Arc::new(move || {
+                                        let addr = rewrite_tunnel.local_addr();
+                                        BrokerRewrite {
+                                            host: addr.ip().to_string(),
+                                            port: Some(addr.port()),
+                                        }
+                                    }),
+                                );
+
+                                let addr = ssh_tunnel.local_addr();
+                                BrokerAddr {
+                                    host: addr.ip().to_string(),
+                                    port: addr.port().to_string(),
+                                }
+                            }
+                            Err(e) => {
+                                // Ensure a log line shows up.
+                                self.error(
+                                    KafkaError::ClientCreation(e.to_string()),
+                                    &e.to_string(),
+                                );
+                                // We have to give rdkafka an address, as this callback can't fail,
+                                // we just give it a random one.
+                                BrokerAddr {
+                                    host: "failed".to_string(),
```

Comment:
We should probably use a domain we control. E.g., `failed-ssh-tunnel.dev.materialize.com`. That way we can guarantee that this hits an NXDOMAIN error.